src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -2552,4 +2552,49 @@ object FunctionRegistry {
         result: [4, 6]
     """, "i" -> "Starting index of the slice.", "j" -> "End index of the slice (not included in result)."
   )(arrayHr(TTHr), intHr, intHr, arrayHr(TTHr))
+
+  registerMethod("[:]", (a: String) => a,
+    """
+    Returns a copy of the string.
+
+    .. code-block:: text
+        :emphasize-lines: 2
+
+        let a = "abcd" in a[:]
+        result: "abcd"
+    """
+  )
+  registerMethod("[*:]", (a: String, i: Int) => a.slice(i, a.length),
+    """
+    Returns a slice of the string from the i*th* unicode-codepoint (0-indexed) to the end.
+
+    .. code-block:: text
+        :emphasize-lines: 2
+
+        let a = "abcdef" in a[3:]
+        result: "def"
+    """, "i" -> "Starting index of the slice."
```

Comment:
This behavior is different than the python behavior with negative indices. I think we need to make it consistent with python.

```python
>>> x = "hailteam"
>>> x[-1:]
'm'
>>> x[-5:]
'team'
```

```scala
>>> val x = 'hailteam'
>>> x.slice(-1, x.length)
String = hailteam
>>> x.slice(-5, x.length) 
String = hailteam
```